accounting/invoice/migrations/0006_auto_20180116_1745.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.8 on 2018-01-16 17:45
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('invoice', '0005_auto_20180104_1203'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='historicalinvoice',
+            name='auto_create_pdf_on_save',
+        ),
+        migrations.RemoveField(
+            model_name='historicalinvoice',
+            name='auto_download_jira_worklogs_on_save',
+        ),
+        migrations.RemoveField(
+            model_name='historicalinvoice',
+            name='auto_upload_google_drive_on_save',
+        ),
+        migrations.RemoveField(
+            model_name='invoice',
+            name='auto_create_pdf_on_save',
+        ),
+        migrations.RemoveField(
+            model_name='invoice',
+            name='auto_download_jira_worklogs_on_save',
+        ),
+        migrations.RemoveField(
+            model_name='invoice',
+            name='auto_upload_google_drive_on_save',
+        ),
+        migrations.AddField(
+            model_name='historicalinvoice',
+            name='approved',
+            field=models.CharField(choices=[('automatically', 'Automatically'), ('manually', 'Manually'), ('not_approved', 'Not Approved')], default='not_approved', help_text='The approval status of this invoice.', max_length=80),
+        ),
+        migrations.AddField(
+            model_name='invoice',
+            name='approved',
+            field=models.CharField(choices=[('automatically', 'Automatically'), ('manually', 'Manually'), ('not_approved', 'Not Approved')], default='not_approved', help_text='The approval status of this invoice.', max_length=80),
```

Comment:
I had gone with a BooleanField before, but @antoviaque wanted to be able to track whether it was approved manually or automatically. I then kept the names succinct because reading `invoice.approved == automatically` feels more right than `invoice.approved = automatically_approved`, which reads with some redundancy IMO.

Let's stick with this for now? I'd have to change I think a lot of places in the code for this and fix up migrations on stage.

Reply:
@UmanShahzad Hm, I agree that `invoice.approved == automatically` reads well... but personally don't mind `invoice.approved = automatically_approved` too much.

The main thing for me is that it feels weird to mix "automatically"/"manually"/"not approved" -- it's like the field can't decide whether it wants to be a Boolean or not ;) "automatically approved"/"manually approved"/"not approved" seems to make a tad more sense as a triple of acceptable values.

I'm fine with covering this as part of a separate ticket though. We can also let @antoviaque weigh in first (i.e., before creating a ticket or adding this item to an upcoming ticket that already exists), to see if he has a strong preference one way or another.